wagtail/contrib/postgres_search/backend.py

```diff
@@ -192,106 +247,136 @@ def __str__(self):
+
         if LOOKUP_SEP in field_lookup:
             field_lookup, sub_field_name = field_lookup.split(LOOKUP_SEP, 1)
         else:
             sub_field_name = None
+
         for field in fields:
-            if isinstance(field, SearchField) \
-                    and field.field_name == field_lookup:
+            if isinstance(field, SearchField) and field.field_name == field_lookup:
                 return field
+
             # Note: Searching on a specific related field using
             # `.search(fields=…)` is not yet supported by Wagtail.
             # This method anticipates by already implementing it.
-            if isinstance(field, RelatedFields) \
-                    and field.field_name == field_lookup:
+            if isinstance(field, RelatedFields) and field.field_name == field_lookup:
                 return self.get_search_field(sub_field_name, field.fields)
 
-    def build_tsquery_content(self, query, group=False):
+    def build_tsquery_content(self, query, invert=False):
         if isinstance(query, PlainText):
-            query_formats = []
-            query_params = []
-            for word in query.query_string.split():
-                query_formats.append(self.TSQUERY_WORD_FORMAT)
-                query_params.append(word)
-            operator = self.TSQUERY_OPERATORS[query.operator]
-            query_format = operator.join(query_formats)
-            if group and len(query_formats) > 1:
-                query_format = '(%s)' % query_format
-            return query_format, query_params
-        if isinstance(query, Boost):
-            return self.build_tsquery_content(query.subquery)
-        if isinstance(query, Not):
-            query_format, query_params = \
-                self.build_tsquery_content(query.subquery, group=True)
-            return '!' + query_format, query_params
-        if isinstance(query, (And, Or)):
-            query_formats = []
-            query_params = []
-            for subquery in query.subqueries:
-                subquery_format, subquery_params = \
-                    self.build_tsquery_content(subquery, group=True)
-                query_formats.append(subquery_format)
-                query_params.extend(subquery_params)
-            operator = (self.TSQUERY_AND if isinstance(query, And)
-                        else self.TSQUERY_OR)
-            return operator.join(query_formats), query_params
+            terms = query.query_string.split()
+            if not terms:
+                return None
+
+            last_term = terms.pop()
+
+            lexemes = Lexeme(last_term, invert=invert, prefix=self.LAST_TERM_IS_PREFIX)
+            for term in terms:
+                new_lexeme = Lexeme(term, invert=invert)
+
+                if query.operator == 'and':
+                    lexemes &= new_lexeme
+                else:
+                    lexemes |= new_lexeme
+
+            return lexemes
+
+        elif isinstance(query, Boost):
```

Comment:
If boosts aren't supported here, should we raise a warning?

Reply:
Added this, thanks!